examples/AdvancedOperations/AddAdCustomizer.php

```diff
@@ -0,0 +1,529 @@
+        return $feedDetails;
+    }
+
+    /**
+     * Creates a feed mapping for a given feed.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int customerId the customer ID
+     * @param array feedDetails the names and IDs of feed attributes
+     */
+    private static function createAdCustomizerMapping(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adCustomizerFeedResourceName,
+        array $feedDetails
+    ) {
+        // Maps the feed attribute IDs to the field ID constants.
+        $nameFieldMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => $feedDetails['Name'],
+            'ad_customizer_field' => AdCustomizerPlaceholderField::STRING
+        ]);
+
+        $priceFieldMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => $feedDetails['Price'],
+            'ad_customizer_field' => AdCustomizerPlaceholderField::PRICE
+        ]);
+
+        $dateFieldMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => $feedDetails['Date'],
+            'ad_customizer_field' => AdCustomizerPlaceholderField::DATE
+        ]);
+
+        // Creates the feed mapping.
+        $feedMapping = new FeedMapping([
+            'placeholder_type' => PlaceholderType::AD_CUSTOMIZER,
+            'feed' => $adCustomizerFeedResourceName,
+            'attribute_field_mappings' => [$nameFieldMapping, $priceFieldMapping, $dateFieldMapping]
+        ]);
+
+        // Creates the operation.
+        $feedMappingOperation = new FeedMappingOperation();
+        $feedMappingOperation->setCreate($feedMapping);
+
+        // Issues a mutate request to add the feed mapping.
+        $feedMappingServiceClient = $googleAdsClient->getFeedMappingServiceClient();
+        $response = $feedMappingServiceClient->mutateFeedMappings(
+            $customerId,
+            [$feedMappingOperation]
+        );
+
+        // Displays the results.
+        foreach ($response->getResults() as $result) {
+            printf(
+                "Created feed mapping with resource name '%s'.%s",
+                $result->getResourceName(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Creates two different feed items to enable two different ad customizations.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param array $feedDetails the names and IDs of feed attributes
+     * @param string $dsaPageUrlLabel the label for the DSA page URLs
+     */
+    private function createFeedItems(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adCustomizerFeedResourceName,
+        array $adCustomizerFeedAttributes
+    ) {
+        $feedItemOperations = [];
+        
+        $marsDate = new DateTime('first day of this month');
+        $feedItemOperations []= self::createFeedItemOperation(
```

Comment:
This is weird.
It should be "array[] =" not "array []="

Reply:
Done.